apps/HelloBaremetal/cmake-twice/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.22)
+
+# Enable assembly language (.s) support additionally
+project(HelloBaremetal LANGUAGES C CXX ASM)
+
+# Set up C++ language settings
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+
+if (BAREMETAL)
+    # Avoid handling Threads library as mandatory dependency in find_package()
+    set(Halide_NO_THREADS ON)
+endif()
+
+if (NOT DEFINED GEN_PACKAGE)
+    set(GEN_PACKAGE "HelloBaremetal-add_generator")
+endif()
+
+# Find Halide. Instead of the package Halide, HalideHelpers is enough as we don't use JIT mode.
+find_package(HalideHelpers REQUIRED)
```

Comment:
As you've observed here and as I've described elsewhere, I think the final design I want to implement (not a blocker here) is to make `find_package(Halide REQUIRED)` do what `find_package(HalideHelpers REQUIRED)` does today and if users know they want JIT mode they can add that as a component as in `find_package(Halide REQUIRED JIT)`.